### PR TITLE
Fix wlroots symbol collision

### PR DESCRIPTION
**Pull request by kennylevinsen**
As plans are to temporarily revert our solution to symbol collisions (https://github.com/swaywm/wlroots/pull/2980), we'll have to instead work around the problem locally.

If we make a 1.6.1 for wlroots 0.14 support, we should cherry-pick this PR as well.

### CODE REVIEW

File: .builds/alpine.yml [hunk 1]
```diff
@@ -38,3 +38,9 @@ tasks:
       cd sway
       meson configure build -Dxwayland=disabled
       ninja -C build
+  - build-static: |
+      cd sway
+      mkdir subprojects
+      ln -s ../../wlroots subprojects/wlroots
+      meson configure build --default-library=static
```
Comment (emersion) [inline .builds/alpine.yml:45]: I think we need to force Meson to use the subproject, or else it'll pick up the one installed system-wide.

Comment (kennylevinsen) [reply .builds/alpine.yml]: It appears that `meson configure` was unable to change dependency source, even with force flags. Now we wipe the build dir and set it up again, with a force-fallback for good measure.

Comment (emersion) [reply .builds/alpine.yml]: Was `meson configure build/ --cleancache` not enough?

Comment (emersion) [reply .builds/alpine.yml]: It doesn't seem like it's enough.

Comment (kennylevinsen) [reply .builds/alpine.yml]: Same conclusion in my attempts (https://builds.sr.ht/~kennylevinsen/job/531097#bottom). :/

**Review by emersion — APPROVED** [review]
Not super happy with re-building the whole wlroots to switch to static libs. Opened https://github.com/mesonbuild/meson/issues/8922 to see if we can easily improve this without having to introduce a custom Sway option.

But this is good enough for now. Thanks!

**Pull request merged**